### PR TITLE
Backport #1213: update pathinfo when changing uri

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Comment by rossabaker** [comment]
A consequence of this I didn't realize: we now have overloads of `withBody`, one of which returns a `Message` and one that returns a `Task[Message]`.  That's ... not great.  If anyone has a clever name or way out of that, it would be great to hear and we could fix it here and bubble it back up to master.

**Comment by rossabaker** [comment]
I added a `withBodyStream` as the low-level method that doesn't deal with payload headers.  The functionality is needed, but I am not fond of the name.  I do strongly think this should not be an overload, as the functionality is quite different.

File: core/src/main/scala/org/http4s/Headers.scala [hunk 1]
```diff
@@ -95,6 +96,14 @@ final class Headers private (headers: List[Header])
       this.toList.equals(otherheaders.toList)
     case _ => false
   }
+
+  /** Removes the `Content-Length`, `Content-Range`, `Trailer`, and
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/Headers.scala]: This PR contains both two indentation and single indentation function annotation style. Should probably decide which style we prefer, or atleast stay consistent within a PR.

Comment (rossabaker) [inline core/src/main/scala/org/http4s/Headers.scala]: Let's default to the scaladoc style.  I anticipate we'll adopt scalafmt, and that would be one less default to change.

**Pull request merged**